rajawali/src/main/java/c/org/rajawali3d/sceneview/SceneView.java

```diff
@@ -0,0 +1,227 @@
+    @NonNull
+    private Rect viewport;
+
+    /**
+     * The {@link Camera} currently being used to render the scene.
+     */
+    @GuardedBy("nextCameraLock")
+    private Camera currentCamera;
+
+    private Camera nextCamera; // The camera the scene should switch to on the next frame.
+    private final Object nextCameraLock = new Object(); // Camera switching lock
+
+    @NonNull
+    @GLThread
+    private Matrix4 viewProjectionMatrix = new Matrix4();
+
+    @Nullable
+    @GLThread
+    private ObjectRenderer lastUsedRenderer; // Reference to the last used object engine
+
+    @Nullable
+    Lock currentlyHeldWriteLock;
+    @Nullable
+    Lock currentlyHeldReadLock;
+
+    // TODO Not sure about volatile semantics here
+    private volatile boolean isDisabled;
+
+    public SceneView(@NonNull Rect viewport) {
+        this.viewport = viewport;
+    }
+
+    public SceneView(@NonNull Scene scene) {
+        this.viewport = new Rect();
+        this.scene = scene;
+    }
+
+    public SceneView(@NonNull Rect viewport, @NonNull Scene scene) {
+        this.viewport = new Rect(viewport);
+    }
+
+    @GLThread
+    public void setEngine(@Nullable Engine engine) {
+        if (engine != null) {
+            onEngineSet(engine);
+        } else {
+            onEngineCleared();
+        }
+    }
+
+    @Nullable
+    @Override
+    public RenderModel getRenderModel() {
+        return scene;
+    }
+
+    @Override
+    public void setViewport(Rect viewport) {
+        this.viewport.set(viewport);
+    }
+
+    @Override
+    public Rect getViewport() {
+        return viewport;
+    }
+
+    @Override
+    public int getViewportWidth() {
+        return viewport.width();
+    }
+
+    @Override
+    public int getViewportHeight() {
+        return viewport.height();
+    }
+
+    @Override
+    public void isDisbled(boolean disabled) {
+        this.isDisabled = disabled;
+    }
+
+    @Override
+    public boolean isDisabled() {
+        return false;
+    }
+
+
+    @GLThread
+    @Override
+    public void onFrameStart(final double deltaTime)
+            throws InterruptedException {
+        // TODO is a scene lock actually needed here?  This may all be independent of the Scene...
+        // TODO is a sceneView lock needed here?
+        //currentlyHeldReadLock = scene.acquireReadLock();
+        try {
+            // TODO frame tasks for changes to viewport and/or render passes
+            // Execute frame tasks
+            //performFrameTasks();
+
+            // TODO view-specific frame callbacks could be useful, e.g. enabling viewport animations
+        } finally {
+            if (currentlyHeldReadLock != null) {
+                currentlyHeldReadLock.unlock();
+            }
+        }
+    }
+
+    @GLThread
+    @Override
+    public void onRenderView() throws InterruptedException {
+        // TODO Lock should probably be on the whole Scene, not just its Graph
```

Comment:
I'm just wondering if other elements in the Scene can change during rendering, e.g. materials?  Or is that already handled somehow? Not totally clear on the locking strategy yet